Kernel/KerrGeoOrbit.m

```diff
@@ -1236,41 +1239,41 @@
-If[e > 1, method = "Analytic"];
-
-If[method == "FastSpec",
-
-	If[param == "Mino",  If[PossibleZeroQ[a] || PossibleZeroQ[e], Return[KerrGeoOrbitMino[a, p, e, x, initPhases]], Return[KerrGeoOrbitFastSpec[a, p, e, x, initPhases]]]];
-	If[param == "Darwin", 
-		If[PossibleZeroQ[a], Return[KerrGeoOrbitSchwarzDarwin[p, e]], Return[KerrGeoOrbitFastSpecDarwin[a,p,e,x,initPhases]]]
-	];
-	Message[KerrGeoOrbit::parametrization, "Unrecognized parametrization: " <> OptionValue["Parametrization"]];
-	
-];
-
-If[method == "Analytic",
-(*Changed "KerrGeoOrbitDarwin" to "KerrGeoOrbitEquatorialDarwin"*)
-	If[param == "Mino", Return[KerrGeoOrbitMino[a, p, e, x, initPhases]]];
-	If[param == "Phases", Return[KerrGeoOrbitPhases[a, p, e, x]]];
-	If[param == "Darwin", 
-		If[PossibleZeroQ[a], Return[KerrGeoOrbitSchwarzDarwin[p, e]], Return[KerrGeoOrbitEquatorialDarwin[a,p,e,x,initPhases]]]
-	];
-	Message[KerrGeoOrbit::parametrization, "Unrecognized parametrization: " <> OptionValue["Parametrization"]];
-
-];
-
-Message[KerrGeoOrbit::general, "Method " <> method <> " is not one of {FastSpec, Analytic}"];
-
-]
+KerrGeoOrbit[a_, p_, e_, x_, initPhases : {_, _, _, _} : {0, 0, 0, 0}, OptionsPattern[]] := Module[{param, method},
+  (*FIXME: add stability check but make it possible to turn it off*)
+  If[a!=0 ||e != 0,If[KerrGeodesics`SpecialOrbits`Private`KerrGeoOnSeparatrixQ[a, p, e, x],Message[KerrGeoOrbit::OnSeparatrix];Abort[];]];
+     method = OptionValue["Method"];
+   param = OptionValue["Parametrization"]; 
+   
+   If[param == "Darwin" && Abs[x] != 1, Message[KerrGeoOrbit::parametrization, "Darwin parameterization only valid for equatorial motion"]; Return[];];
+   
+   If[Precision[{a, p, e, x}] > 30, method = "Analytic"];
+   If[e > 1, method = "Analytic"];
+   
+   If[method == "FastSpec",
+    
+    	If[param == "Mino",  If[PossibleZeroQ[a] || PossibleZeroQ[e], Return[KerrGeoOrbitMino[a, p, e, x, initPhases]], Return[KerrGeoOrbitFastSpec[a, p, e, x, initPhases]]]];
+    	If[param == "Darwin", 
+     		If[PossibleZeroQ[a], Return[KerrGeoOrbitSchwarzDarwin[p, e]], Return[KerrGeoOrbitFastSpecDarwin[a, p, e, x, initPhases]]]
+     	];
+    	Message[KerrGeoOrbit::parametrization, "Unrecognized parametrization: " <> OptionValue["Parametrization"]];
+    	
+    ];
+   
+   If[method == "Analytic",
+    (*Changed "KerrGeoOrbitDarwin" to "KerrGeoOrbitEquatorialDarwin"*)
+    	If[param == "Mino", Return[KerrGeoOrbitMino[a, p, e, x, initPhases]]];
+    	If[param == "Phases", Return[KerrGeoOrbitPhases[a, p, e, x]]];
+    	If[param == "Darwin", 
+     		If[PossibleZeroQ[a], Return[KerrGeoOrbitSchwarzDarwin[p, e]], Return[KerrGeoOrbitEquatorialDarwin[a, p, e, x, initPhases]]]
+     	];
+    	Message[KerrGeoOrbit::parametrization, "Unrecognized parametrization: " <> OptionValue["Parametrization"]];
+    
+    ];
+   
+   Message[KerrGeoOrbit::general, "Method " <> method <> " is not one of {FastSpec, Analytic}"];
+   
+   ]
```

Comment:
Lots of unrelated whitespace changes here...

Reply:
Not sure why these all changed. I didn't manually add these in so it must have been automatic.